File: prravda/src/theater-booking-application/discount-policies/amount-discount-policy.ts
import { AbstractDiscountPolicy } from "./abstract.discount-policy";
import { DiscountCondition } from "../discount-conditions/discount-condition";
import { Screening } from "../screening";
import { Money } from "../money";

export class AmountDiscountPolicy extends AbstractDiscountPolicy {
  constructor(private discountAmount: Money, conditions: DiscountCondition[]) {
    super(conditions);
  }

  protected getDiscountAmount(screening: Screening): Money {
    return this.discountAmount;
  }
}

File: jinia/object/src/main/java/chapter2/application/AmountDiscountPolicy.java
package chapter2.application;

import chapter2.model.Money;
import chapter2.model.Screening;

public class AmountDiscountPolicy extends DiscountSkeletonPolicy{
    private Money discountAmount;

    public AmountDiscountPolicy(Money discountAmount, DiscountCondition... conditions) {
        super(conditions);
        this.discountAmount = discountAmount;
    }

    @Override
    protected Money getDiscountAmount(Screening screening) {
        return discountAmount;
    }
}
